### ISSUE TITLE
Huge vertical deviation for a long dataset

### ISSUE THREAD

**Issue by mhaboali** [HKUST-Aerial-Robotics/A-LOAM#35]
Hi thanks for sharing such a good project. 

I've tested your project with my own dataset which is around 1700 seconds. What I noticed there was a huge vertical deviation for the path after around 1300 seconds as shown below

**allMapCloud**
![image](https://user-images.githubusercontent.com/29764281/83003230-51c3e380-a00e-11ea-9c25-7a5f8a9b22de.png)

**AftMapped Path**
It's very clear here, that's supposed to be at the same level.
![image](https://user-images.githubusercontent.com/29764281/83003720-ff36f700-a00e-11ea-8515-40b7b845df80.png)

And the ground truth doesn't have this vertical deviation at all, all streets were at the same level.

What do you think the issue would come from? And how can I resolve this issue?

Thanks!

**Comment by mhaboali** [comment]
Hi @qintonguav,
Please let me know what do you think of this issue?

Thanks!

**Comment by gisbi-kim** [comment]
@mhaboali 
Actually, for the large-scale env, the phenomenon is inevitable I think. You can consider loop detection and loop closing of a pose-graph (e.g., https://github.com/irapkaist/SC-LeGO-LOAM could be a solution)

**Comment by mhaboali** [comment]
Thanks @kissb2 for your recommendation, I'll try it shortly and I'll keep you posted on the status.

Thanks again and I appreciate your help!
Mohammad

**Comment by EdCheriee** [comment]
I have a similar issue on larger datasets (beyond 1200s) and have the same issue as yours and doesn't detect the height levels right (when returning to the same point it doesn't recognize the same ground level). Let me know if you had any success (trying it myself as well).

**Comment by mhaboali** [comment]
Yeah @EdCheriee , my dataset is around 1700 seconds. Actually, I didn't manage to resolve this issue with A-Loam but I'm still searching for another alternative. Let's keep the issue here updated with our findings and let's hope we'll find a proper alternative.

**Comment by papicheng** [comment]
帮顶，同遇到这个问题，难道原作者是忽略这个问题了吗[笑脸]
